core/src/main/java/com/netflix/conductor/auth/AuthManager.java

```diff
@@ -97,6 +101,23 @@ public AuthManager(Configuration config) {
 	}
 
 	public AuthResponse authorize(Workflow workflow) throws Exception {
+		synchronized (this) {
```

Comment:
Can we use double synchronized approach but only if `cachedAuthDataRef.get()` has noting in it?

Reply:
Performance shall be better

Reply:
```
Map<String, Object> cachedAuthData = cachedAuthDataRef.get();
if (isNotExpiredAuthData(cachedAuthData)) {
    return (AuthResponse) cachedAuthData.get("authResponse");
}
synchronized (this) {
Map<String, Object> cachedAuthData = cachedAuthDataRef.get(); // read again
// check again
and then only continue 

AuthResponse authResponse = executeAuthorization(workflow);

		if (authResponse.getError() == null && authResponse.getErrorDescription() == null) {
			cacheAuthData(authResponse);
		}
		return authResponse;
}

```